src/test/java/hudson/tasks/junit/JUnitResultArchiverTest.java

```diff
@@ -96,20 +98,24 @@
 public class JUnitResultArchiverTest {
 
     @Rule public JenkinsRule j = new JenkinsRule();
+
+    @ClassRule
+    public final static BuildWatcher buildWatcher = new BuildWatcher();
+
     private FreeStyleProject project;
     private JUnitResultArchiver archiver;
 
     @Before public void setUp() throws Exception {
         project = j.createFreeStyleProject("junit");
         archiver = new JUnitResultArchiver("*.xml");
         project.getPublishersList().add(archiver);
-
         project.getBuildersList().add(new TouchBuilder());
     }
 
     @LocalData("All")
     @Test public void basic() throws Exception {
-        FreeStyleBuild build = project.scheduleBuild2(0).get(10, TimeUnit.SECONDS);
+        FreeStyleBuild build =
+                j.assertBuildStatus(Result.UNSTABLE, j.waitForCompletion(project.scheduleBuild2(0).waitForStart()));
```

Comment:
Is this related somehow or just general cleanup that could be in another PR?

Reply:
just removing a pattern which can be flaky